frontend/src/pages/learningCenter/ProviderTypeFilters.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { useHistory } from 'react-router';
+import { OdhDocument } from '../../types';
+import {
+  FilterSidePanelCategory,
+  FilterSidePanelCategoryItem,
+} from '@patternfly/react-catalog-view-extension';
+import { removeQueryArgument, setQueryArgument } from '../../utilities/router';
+import { PROVIDER_TYPE_FILTER_KEY } from './const';
+import { useQueryFilters } from './useQueryFilters';
+
+type ProviderTypeFiltersProps = {
+  docApps: OdhDocument[];
+  categoryApps: OdhDocument[];
+};
+
+const ProviderTypeFilters: React.FC<ProviderTypeFiltersProps> = ({ docApps, categoryApps }) => {
+  const history = useHistory();
+  const providerTypeFilters = useQueryFilters(PROVIDER_TYPE_FILTER_KEY);
+  const [showAll, setShowAll] = React.useState<boolean>(false);
+
+  const providerTypes = React.useMemo(() => {
+    const allTypes = {};
+    docApps.forEach((docApp) => {
+      if (!docApp.spec.appCategory) {
+        return;
+      }
+      allTypes[docApp.spec.appCategory] = 0;
+    });
+    categoryApps.forEach((categoryApp) => {
+      if (!categoryApp.spec.appCategory) {
+        return;
+      }
+      allTypes[categoryApp.spec.appCategory]++;
```

Comment:
question: concerned about `NaN` if `categoryApp.spec.appCategory` isn't in `allTypes` ?

Reply:
the category apps are a filtered set of all apps, so that should never happen.

Reply:
Added a check just to be safe